astropy/coordinates/velocities.py

```diff
@@ -0,0 +1,74 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Tools for computing velocities and velocity corrections using Astropy
+coordinates and related machinery.
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from .. import units as u
+
+from .solar_system import get_body_barycentric_posvel
+from .matrix_utilities import matrix_product
+from .representation import CartesianRepresentation, UnitSphericalRepresentation
+from .builtin_frames import GCRS
+
+
+__all__ = ['helio_corr', 'helio_vector']
+
+KPS = u.km/u.s
+
+def helio_vector(t, loc):
+    """
+    Compute the heliocentric velocity correction at a given time and place.
+
+    Paramters
+    ---------
+    t : astropy.time.Time
+        Time of the observation. Can be a Time array.
+    loc : astropy.coordinates.EarthLocation
+        The observer location at which to compute the correction.
+    """
+    vsun = get_body_barycentric_posvel('sun', t)[1]
+    vearth = get_body_barycentric_posvel('earth', t)[1]
+
+    vsunearth = vearth - vsun
+
+    gcrs_p, gcrs_v = loc.get_gcrs_posvel(t)
+
+    vsuntargxyz = vsunearth.xyz + gcrs_v.xyz
+    return CartesianRepresentation(vsuntargxyz.to(KPS))
+
+
+def helio_corr(t, loc, target):
```

Comment:
We had a long discussion for the barycentric time corrections about where these things should live. For velocity corrections, to me `SkyCoord` is more logical, in which the signature would become `(SkyCoord, Time, Location)` - with defaults of `sc.obstime` and `sc.location` (and, if a time is passed in, a secondary default of `time.location`).

That said, I think it would be fine to have a short-cut on `Time` (and similarly a short-cut for time corrections on `SkyCoord`).